pkg/registry/core/service/storage/rest.go

```diff
@@ -1176,3 +1190,61 @@ func collectServiceNodePorts(service *api.Service) []int {
 	}
 	return servicePorts
 }
+
+// nearlyNullStrategy does almost nothing, but makes the REST logic happy.
+type nearlyNullStrategy struct {
+	runtime.ObjectTyper
+	names.NameGenerator
+}
+
+// NamespaceScoped is true for services.
+func (nearlyNullStrategy) NamespaceScoped() bool {
+	return true
+}
+
+func (nearlyNullStrategy) PrepareForCreate(_ context.Context, _ runtime.Object) {
+}
+
+func (nearlyNullStrategy) PrepareForUpdate(_ context.Context, _, _ runtime.Object) {
+}
+
+func (nearlyNullStrategy) Validate(_ context.Context, _ runtime.Object) field.ErrorList {
```

Comment:
@thockin the unit tests fail now because we no longer validate here
https://github.com/kubernetes/kubernetes/blob/d01cc01ab4455d1c0db84d2cc79d963a1b15d292/pkg/registry/core/service/strategy.go#L125-L130

should we modify the test or the strategy?

Reply:
The testing here is ALL OVER THE PLACE.  Before this change it tested with strategy, but strategy has its own tests.  Some of the test in here only test validation, which also has its own tests.  There are some general-looking (i.e. follows patterns) tests for the inner REST but the failing test are the outer rest.

So the question at hand is - should the UNIT TEST for the "outer" REST test all the way down into the inner REST?  It never has.  I can add strategy back in here,  but it feels sketchy to be testing across packages that way.  Or I can remove the validation-centric tests and just make this pass without strategy, then come back as we clean up the REST logic overall, which is where I am leaning.